src/Hungary/Mews.Fiscalizations.Hungary/Models/Invoice/ForeignCompany.cs

```diff
@@ -0,0 +1,36 @@
+﻿using FuncSharp;
+using Mews.Fiscalizations.Core.Model;
+using System.Collections.Generic;
+
+namespace Mews.Fiscalizations.Hungary.Models
+{
+    public sealed class ForeignCompany
+    {
+        private ForeignCompany(Name name, SimpleAddress address, TaxpayerIdentificationNumber taxpayerId = null)
+        {
+            Name = name;
+            Address = address;
+            TaxpayerId = taxpayerId.ToOption();
+        }
+
+        public Name Name { get; }
+
+        public SimpleAddress Address { get; }
+
+        public IOption<TaxpayerIdentificationNumber> TaxpayerId { get; }
+
+        public static ITry<ForeignCompany, IEnumerable<Error>> Create(Name name, SimpleAddress address, TaxpayerIdentificationNumber taxpayerId = null)
+        {
+            var optionalForeignTaxPayerId = taxpayerId.ToOption().ToOption().Where(i => i.Match(
+                identifier => identifier.Country.Equals(Countries.Hungary),
+                _ => true
+            ));
+            return Try.Aggregate(
+                optionalForeignTaxPayerId.ToTry(_ => Error.Create($"{nameof(TaxpayerIdentificationNumber)} must be a foreign (non-Hungarian) taxpayer number.")),
+                ObjectValidations.NotNull(name),
+                ObjectValidations.NotNull(address),
```

Comment:
Didn't we say we will avoid these `NotNull` validations because they are not practical? I guess it doesn't really matter if there are other validations and the return type is already an ITry, but it can simplify methods that would otherwise have a simple return type. But overall, I think it should be consistent everywhere.

cc @KaliCZ

Reply:
We said that we will not write the Create methods returning an ITry if all it is checking are nulls. But if we already have a create method, we will check the nulls, because if we have a method returning an ITry, it should be complete.